### ISSUE TITLE
Add support for APNG when libspng adds it

### ISSUE THREAD

**Issue by jcupitt** [libvips/libvips#2537]
### Discussed in https://github.com/libvips/libvips/discussions/2000

<div type='discussions-op-text'>

<sup>Originally posted by **JGVB** November 21, 2016</sup>
APNG's are on the rise and I am wondering if there is support for them in this library?  Or if there could be with any modifications?</div>

**Comment by blipk** [comment]
Might as well add VP9 WebM support.

**Comment by kleisauke** [comment]
Issue https://github.com/randy408/libspng/issues/4 is relevant here.

**Comment by jcupitt** [comment]
We have a grant from nlnet to add this: 

https://nlnet.nl/project/libvips/

libspng too, so we should coordinate on this over the next few months.

**Comment by sturkmen72** [comment]
> We have a grant from nlnet to add this:
> 
> https://nlnet.nl/project/libvips/
> 
> libspng too, so we should coordinate on this over the next few months.

i am eager to work on this issue if needed. i already worked on adding apng support to OpenCV as GSOC 2024 project

**Comment by sturkmen72** [comment]
afaik @randy408 working on adding apng support to libspng but it seems there is no activity on libspng.

**Comment by jcupitt** [comment]
Yes, I've parked that part of the nlnet grant for now. I plan to look at improving HDR support instead, working on libjxl first.

**Comment by jcupitt** [comment]
See https://github.com/pnggroup/libpng/issues/267

libpng 1.8, now in beta, should have APNG support.

See also https://github.com/libvips/vipsdisp/issues/59
